source/SH3/system/glprogram.cpp

```diff
@@ -75,11 +80,17 @@ GLuint sh3_glprogram::Compile(std::uint32_t type)
 
         source.open(fname);
 
+        if(!source.is_open())
+            die("sh3_glprogram::Compile( ): Unable to open a handle to %s!", fname.c_str());
+
         while(!source.eof())
```

Comment:
We could require the shader sources to have UNIX-style line endings ('\n'), in which case we can open the file in binary mode and use a `std::ostringstream` and read the whole file via `stringstream << source.rdbuf();`.
Reading the file like this is safer though.